server/validation/validation.go

```diff
@@ -15,6 +15,9 @@ func ValidateTuple(ctx context.Context, backend storage.TypeDefinitionReadBacken
 	if err := tuple.ValidateUser(tk); err != nil {
 		return nil, err
 	}
+	if err := ValidateUserIfItsAUserset(ctx, backend, store, authorizationModelID, tk); err != nil {
```

Comment:
Please don't delegate this obscurely like it is doing right now. It's not obvious the top level path this is taking whatsoever. Instead, I recommend we make this top-level code path more explicit with

```
if tuple.IsObjectRelation(tk.GetUser()) {
   ... // validation logic
}
```

this way the code is more inline and obvious what validations are taking place in the event the user field is an ObjectRelation (e.g. userset).